tests/external/notifications/aws/test_aws_notification_handler.py

```diff
@@ -0,0 +1,208 @@
+                   ('x-amz-sns-topic-arn', 'arn:aws:sns:us-west-2:123456789012:MyTopic'),
+                   ('x-amz-sns-subscription-arn', 'arn:aws:sns:us-west-2:123456789012:MyTopic:2bcfbf39-05c3-41de-beaa-fcfcc21c8f55'),
+                   ('Content-Length', 761),
+                   ('Content-Type', 'text/plain; charset=UTF-8'),
+                   ('Host', 'ec2-50-17-44-49.compute-1.amazonaws.com'),
+                   ('Connection', 'Keep-Alive'),
+                   ('User-Agent', 'Amazon Simple Notification Service Agent')]
+
+        body_dict = {"Type" : "Notification",
+                     "MessageId" : "4d9d04a3-113f-5532-8352-57141f47f5c4",
+                     "TopicArn" : "arn:aws:sns:us-east-1:589179999999:CostUsageReportUpdateTopic",
+                     "Subject" : "Amazon S3 Notification",
+                     "Message" : "{\"Records\":[{\"eventVersion\":\"2.0\",\"eventSource\":\"aws:s3\",\"awsRegion\":\"us-east-1\",\"eventTime\":\"2018-07-04T18:04:39.895Z\",\"eventName\":\"ObjectCreated:Put\",\"userIdentity\":{\"principalId\":\"AWS:AIDAJLSI7DTYBUZAOKHVM\"},\"requestParameters\":{\"sourceIPAddress\":\"10.85.145.33\"},\"responseElements\":{\"x-amz-request-id\":\"82DD93AB926CD233\",\"x-amz-id-2\":\"gdJjPizP/oLKRROOFBUDxgxanwSM9f6PQyvzP0MaJlhBZ9LCB+7sj6z70GIuAoKT5zfx4soso4U=\"},\"s3\":{\"s3SchemaVersion\":\"1.0\",\"configurationId\":\"CostUsageNotification\",\"bucket\":{\"name\":\"cost-usage-bucket\",\"ownerIdentity\":{\"principalId\":\"A2V75OLLZ1ABF7\"},\"arn\":\"arn:aws:s3:::cost-usage-bucket\"},\"object\":{\"key\":\"/koku/20180701-20180801/bb976103-7c20-4053-852c-ad0c4b778dd0/koku-1.csv.gz\",\"size\":53641,\"eTag\":\"57b759ec6f3bf98bafa3991f5b9bac15\",\"sequencer\":\"005B3D0C37CE41DED4\"}}}]}",
+                     "Timestamp" : "2018-07-04T18:04:40.036Z",
+                     "SignatureVersion" : "1",
+                     "Signature" : "gMXR9TvaiZjpuslyjnfj9PdOyOg78qqrz4pyt7Ct7i+EWkNbo1ePKoszJv2Wx0k7joX+irfLiCiROt8lr3TuVEA9ht/lG1hQAUW6T+BAikB6IF5c9uR1PtiP+Kn+qG9FyDbtSdr3Qn0i3PwI9vOIrqXNf5aEm1mJKqdwdWvkQ+pYHr1RYn4a5T6rrUuRicfzhD6qlbq8jfgzhHPLUcn4wQZ9Sdq7buuY8I3jCo/p+CGhkg70HaF64dyuzSZQU7DEaumBuMvasTI50w3PgymF4CM3wBGrzqEc+FFoEAgUjJTO2ri//KfBzhMhUkxhkIzzh5rGgrHfVGpV7gyFJ9vGWw==",
+                     "SigningCertURL" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-eaea6120e66ea12e88dcd8bcbddca752.pem",
+                     "UnsubscribeURL" : "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:589179999999:CostUsageReportUpdateTopic:7d549a1b-502f-47f7-9469-9ed18671e76a"}
+        body = json.dumps(body_dict)
+        handler = AWSNotificationHandler(headers, body)
+        with self.assertRaises(NotificationInterfaceFilter):
+            _ = handler.get_billing_source()
+
+    def test_get_billing_source_empty_body(self):
+        headers = [('X-Amz-Sns-Message-Type', 'Notification')]
+        body = u''
```

Comment:
IIRC, all strings in Python 3 are unicode. I don't think it matters if you use `''` versus `u''`